pkg/status/status_otlp.go

```diff
@@ -8,22 +8,25 @@
 package status
 
 import (
-	"github.com/DataDog/datadog-agent/cmd/agent/common"
+	"github.com/DataDog/datadog-agent/comp/otelcol/collector"
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/otlp"
 )
 
+// OTLPCollector holds an instance of any running collector.
+// TODO: remove once this package is migrated to components.
+var OTLPCollector collector.Component
```

Comment:
Can we set this to a dummy `collector.Component` on an `init` function? To avoid a crash if we get something wrong re: initialization order

Reply:
No, please let's not add an `init` function as well. Having globals is already bad enough. How do you foresee a crash happening? There is a `nil` check where this is called.

Reply:
Okay, nil check should be enough if there is no possibility of going back to nil. Could this lead to a data race though? Do we need to use an atomic pointer here?

Reply:
My only remaining comment is this one: do we need to wrap this in an atomic pointer?